src/lib/cimb_translator/CimbReader.cpp

```diff
@@ -27,13 +27,16 @@ namespace {
 	template <typename MAT>
 	bitbuffer preprocessSymbolGrid(const MAT& img, bool needs_sharpen)
 	{
-		int blockSize = 3; // default: no preprocessing
+		int blockSize = 5; // default: no preprocessing
 
 		cv::Mat symbols;
 		cv::cvtColor(img, symbols, cv::COLOR_RGB2GRAY);
 		if (needs_sharpen)
-			sharpenSymbolGrid(symbols, symbols); // we used to change blockSize for this case -- may one day be a useful trick again?
-		cv::adaptiveThreshold(symbols, symbols, 255, cv::ADAPTIVE_THRESH_MEAN_C, cv::THRESH_BINARY, blockSize, -5);
+		{
+			blockSize = 7;
+			sharpenSymbolGrid(symbols, symbols);
+		}
+		cv::adaptiveThreshold(symbols, symbols, 255, cv::ADAPTIVE_THRESH_MEAN_C, cv::THRESH_BINARY, blockSize, 0);
```

Comment:
It's not clear the best way to calculate it (do you include good and bad images? borderline ones? etc etc), but the most sensible way I have of evaluating the error rate introduced by that `-5` is to check how many of the sample images fit into certain error cutoffs.

Before and after, using blockSize=3 and different constant values (top=-5, bottom=0):

| <50  | < 100 | < 200 | < 400 | < 500 | <700 | < 1000 | total |
| ---- | ----- | ----- | ----- | ----- | ---- | ------ | ----- |
| 28   | 62    | 98    | 149   | 168   | 190  | 223    |  371  |
| 107  | 141   | 177   | 208   | 226   | 251  | 266    |  371  |

Higher is better. Though for very low error rates the difference is mostly irrelevant (the ECC will correct a 1 error or 60 with similar ease), on the whole this means that the previous change was a significant regression under certain lighting conditions -- the 371 image dataset includes various lighting scenarios -- and fine under others.